client/my-sites/checkout/composite-checkout/lib/paypal-express-processor.ts

```diff
@@ -27,28 +27,32 @@ export default async function payPalProcessor(
 		siteId,
 		siteSlug,
 		contactDetails,
-		customizedCancelUrl,
 	} = transactionOptions;
 	reduxDispatch( recordTransactionBeginAnalytics( { paymentMethodId: 'paypal' } ) );
 
 	const thankYouUrl = getThankYouUrl();
 	let currentUrl;
 	let currentBaseUrl;
+	let currentUrlQuery;
 	try {
 		currentUrl = window.location.href;
 		currentBaseUrl = window.location.origin;
+		currentUrlQuery = window.location.search;
 	} catch ( error ) {
 		currentUrl = `https://wordpress.com/checkout/${ siteSlug }`;
 		currentBaseUrl = 'https://wordpress.com';
+		currentUrlQuery = '';
 	}
 	const currentUrlWithoutQuery = currentUrl.split( /\?|#/ )[ 0 ];
+	const updatedQuery = new URLSearchParams( currentUrlQuery );
 	const successUrl = thankYouUrl.startsWith( 'http' ) ? thankYouUrl : currentBaseUrl + thankYouUrl;
-	let cancelUrl = customizedCancelUrl;
-	if ( ! cancelUrl ) {
-		cancelUrl = createUserAndSiteBeforeTransaction
-			? currentUrlWithoutQuery + '?cart=no-user'
-			: currentUrlWithoutQuery;
+	if ( createUserAndSiteBeforeTransaction ) {
+		updatedQuery.append( 'cart', 'no-user' );
```

Comment:
Use `set` might be better to avoid duplicate key in query string?
```js
updatedQuery.set( 'cart', 'no-user' );
```

Reply:
Oh, brilliant. Thank you. I'm new to the URLSearchParams API!